File: Laborator/Lab 3/JavaClientServer/Services/src/main/java/Services/IService.java
package Services;

import Domain.Child;
import Domain.Event;

import java.util.List;

public interface IService {
    void login(String userName, String Password,IObserver client) throws Error;
    void logout(String userName,IObserver client) throws Error;
    List<Child> getFilteredChildren(int idEvent,int ageMin,int ageMax) throws Error;
    List<Event> getAllEvents() throws Error;
    Child saveChild(Child c) throws Error;
    int countChildren(int idEvent) throws Error;
}

File: Laborator/Lab 3/C# Client-server/Services/IService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;

namespace Services
{
    public interface IService
    {
        void Login(string username, string password, IObserver client);
        void Logout(string username, IObserver client);
        List<Child> GetFilteredChildren(int idEvent, int ageMin, int ageMax);
        List<Event> GetAllEvents();
        Child SaveChild(Child c);
        int CountChildren(int idEvent);
    }
}
